elements/lisk-chain/src/diff.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+type HistoryType = [string?, number?];
```

Comment:
is it optional for string and number?

Reply:
this is optional because when initialize here we have a blank history
`const frontier: Frontier = { 1: { x: 0, history: [] } };`

Reply:
I think empty array should be fine even without the option?

Reply:
strange, earlier it was giving me lint error and now it doesn't, I'll remove optional